zork++/src/lib/utils/template.rs

```diff
@@ -0,0 +1,199 @@
+use crate::config_cli::CppCompiler;
+use log::info;
+use std::io::Write;
+use std::path::PathBuf;
+use std::process::Command;
+use std::str::FromStr;
+use std::{
+    fs::{DirBuilder, File},
+    path::Path,
+};
+
+/// Generates a new C++ standarized empty base project
+///        with a pre-designed structure to organize the
+///        user code in a modern fashion way.
+///        Base design for create the project files and folders:
+///            - ./ifc/<project_name>
+///                - hello_zork.<mod_extension>
+///            - ./src/<project_name>
+///                - hello_zork.cpp
+///            - test
+///            - dependencies
+pub fn create_template_project(
+    name_template: &Option<String>,
+    git: bool,
+    compiler: &Option<CppCompiler>,
+) {
+    // TODO required validate or control optionals
+    info!("Create template project");
+    let name = name_template.as_deref().unwrap_or("calculator");
```

Comment:
Review the lines 29-32. There's no need to dereference the option

Reply:
Its solved